### ISSUE TITLE
Monitor Yugabyte support

### ISSUE THREAD

**Issue by rbygrave** [ebean-orm/ebean#2510]
Monitor https://github.com/yugabyte/yugabyte-db/issues/11186

For those unaware of Yugabyte, it has Postgres compatible types and API - uses Postgres JDBC driver etc (so like Cockroach in this respect).

Features supported: https://github.com/yugabyte/yugabyte-db/blob/master/architecture/YSQL-Features-Supported.md

--------
Issues:
- `create table like ` ... is not supported. Error executing stmt[create table hembi_bean_history(like hembi_bean)] error[ERROR: LIKE clause not supported yet
  Hint: See https://github.com/YugaByte/yugabyte-db/issues/1129. Click '+' on the description to raise its priority
  Position: 33]
- ERROR: column "plain_bean" is of type json but expression is of type character varying   Hint: You will need to rewrite or cast the expression.
- column "cidr" is of type cidr but expression is of type character varying   Hint: You will need to rewrite or cast the expression.
- ERROR: function pgp_sym_encrypt(character varying, character varying) does not exist   Hint: No function matches the given name and argument types. You might need to add explicit type casts.

**Comment by rbygrave** [comment]
Add support in ebean-test-docker for yugabyte docker - https://github.com/ebean-orm/ebean-test-docker/pull/64

**Comment by rbygrave** [comment]
Add yugabyte as platform defined in ebean-annotation Platform - https://github.com/ebean-orm/ebean-annotation/issues/50

**Comment by rbygrave** [comment]
Initial platform support with DDL generation and Platform detection - https://github.com/ebean-orm/ebean/pull/2516

**Comment by rbygrave** [comment]
Use Postgres JSON/JSONB ScalarTypes (but not HSTORE) - https://github.com/ebean-orm/ebean/pull/2529

**Comment by rbygrave** [comment]
Modify ebean-test to support `ebean.test.platform=yugabyte` - https://github.com/ebean-orm/ebean/pull/2534

**Comment by rbygrave** [comment]
Automated tests against Yugabyte passing (with dedicated github workflow) - https://github.com/ebean-orm/ebean/actions/workflows/yugabyte.yml

**Comment by priyanshi-yb** [comment]
Thanks Ebean team for such quick support! 
@rbygrave As we have published docs on Ebeans ORM example on Yugabyte website ([here](https://docs.yugabyte.com/latest/quick-start/build-apps/java/ysql-ebeans/)).So, Can you please let us know the procedure or any requirements we have to provide from our end to get YugabyteDB listed on ebean site under Database Platforms [here](https://ebean.io/docs/database/)? Thanks!

**Comment by rPraml** [comment]
@priyanshi-yb i took a short look at the documentation and I would recommend to use java 11 instead of java 8. See #2374

**Comment by rbygrave** [comment]
@priyanshi-yb I understand how that example got to be how it is (using Play and sbt and requiring a Yugabyte db as prerequisite etc) but as I see it, I'd like an example that shows how we would expect to **_TEST_** an application (and use maven rather than sbt). That is, when we use `ebean-test` we don't need an existing Yugabyte database but instead one is created and started as necessary using docker - this includes setting up the database, creating a database user etc.

So as I see it there is the need to create an example that is more like what I'd expect in terms of how testing works, uses `ebean-test` (to auto start/create docker version of yugabyte), creates ddl automatically, drops and creates tables when running tests, maybe seeds the database with some test data etc.  I'd like to see that as well as yes creating the documentation on the Ebean side. There isn't anything from stopping that Ebean documentation to be created - just need to action it.


As a tiny point, I see `ebeans` used a lot rather than `ebean` - there is no trailing `s` on `Ebean`.

**Comment by priyanshi-yb** [comment]
@rbygrave I see, we have [this](https://docs.google.com/document/d/1OU9IWgtOkmp13aj6Mn8FJ2PW_diUet4Tg-qmVIbIq58/edit?usp=sharing) doc ready for now which is for how testing works by ebean test suite against database platforms. So, can you please point us to such an example application for other db platforms which we can look at?
And also like where would you like to put such an example application?

**Comment by priyanshi-yb** [comment]
@rbygrave can you please point us to such an example so that we could provide you the same as early as possible? Thanks!

**Comment by rbygrave** [comment]
Well, for example with `ebean-test.yaml` and `platform: yugabyte` in:
https://github.com/ebean-orm/examples/blob/master/base-example/src/test/resources/application-test.yaml#L11

The `ebean-test` will automatically start yugabyte as a docker container, create the db, role, and add extensions like pgcrypto ... and we can see the in the trace logs like:

```
17:36:13.882 [main] INFO  io.ebean - ebean version: 13.0.0
17:36:13.891 [main] INFO  io.avaje.config - Loaded properties from [resource:application.yaml, resource:application-test.yaml] 
17:36:13.913 [ForkJoinPool.commonPool-worker-2] INFO  io.ebean.test - Using jdbc settings - username:my_app9 url:jdbc:postgresql://localhost:6433/my_app driver:org.postgresql.Driver
17:36:14.163 [ForkJoinPool.commonPool-worker-2] TRACE io.ebean.docker.commands.Commands - sqlRun: select 1 from pg_database where datname = 'my_app'
17:36:14.192 [ForkJoinPool.commonPool-worker-2] TRACE io.ebean.docker.commands.Commands - sqlRun: select rolname from pg_roles where rolname = 'my_app9'
17:36:14.207 [ForkJoinPool.commonPool-worker-2] DEBUG io.ebean.docker.commands.Commands - sqlRun: create database my_app with owner my_app9
17:36:19.010 [ForkJoinPool.commonPool-worker-2] DEBUG io.ebean.docker.commands.Commands - sqlRun: create extension if not exists "pgcrypto"
17:36:19.544 [ForkJoinPool.commonPool-worker-2] DEBUG io.ebean.docker.commands.Commands - Container ut_yugabyte ready with host:localhost port:6433
17:36:19.551 [main] INFO  io.ebean.test - For testing purposes a current user and tenant provider has been configured. Use io.ebean.test.UserContext to set current user and tenant in tests.
17:36:19.779 [main] INFO  io.ebean.datasource - DataSourcePool [db] autoCommit[false] transIsolation[READ_COMMITTED] min[2] max[200] in[208ms]
17:36:20.271 [main] INFO  io.ebean.DDL - Executing db-drop-all.sql - 31 statements, autoCommit:true
17:36:20.396 [main] INFO  io.ebean.DDL - Executing db-create-all.sql - 33 statements, autoCommit:false
17:36:25.048 [main] INFO  io.ebean.DDL - Executing extra-ddl - 1 statements, autoCommit:false
17:36:25.118 [main] INFO  io.ebean.core - Started database[db] platform[YUGABYTE] in 5567ms
```

That said, what is **_NOT_** in this example is examples of:
- Best practice for asserting the state of the database after a given action
- Best practice in terms of mapping between "DTOs" and "Entity Beans" ...

**Comment by rbygrave** [comment]
WRT Database migrations there are a bunch of migration tests that do not pass with current Yugabyte.  These are documented via the PR - https://github.com/ebean-orm/ebean/pull/2572 ... which we will not merge at this stage.  

Instead we can review this later with the view that some of these issues will be addressed by Yugabyte. Apps hitting these issues can manually tweak migrations with extra statements as needed to work around these issues.

**Comment by rbygrave** [comment]
@priyanshi-yb The Ebean docs have been updated for Yugabyte now at - https://ebean.io/docs/database/yugabyte/

This includes how to use `ebean-test` to automatically initialise a Yugabyte docker container, plus how to do this programmatically. I have noticed that Yugabyte is relatively slow to execute DDL relative to Postgres.

Anyway, I'll look to close this now. Let me know if there is other documentation improvements we think we need.
